### PR TITLE
Fix allowed rule when submitting a list of unashables.

### LINKED ISSUE #524: allowed rule crashes with unashable values
Used Cerberus version / latest commit: 631e05d9773991238dfd7d994a637887fca348d7

- [x] I have the capacity to improve the docs when my problem is solved.
- [x] I have the capacity to submit a patch when a bug is identified.

- [x] My question does not concern a practical use-case that I can't figure out
      to solve.

----

#### Bug report / Feature request

Validating the following input:

```json
{
  "letters": [{"foo": "bar"}]
}
```

against the following schema:

```json
{
  "letters": {
    "type": "list",
    "allowed": ["a", "b", "c"]
  }
}
```

leads to the following crash:

```
   File "./cerberus/validator.py", line 997, in validate
     self.__validate_definitions(definitions, field)
   File "./cerberus/validator.py", line 1066, in __validate_definitions
     result = validate_rule(rule)
   File "./cerberus/validator.py", line 1041, in validate_rule
     return validator(definitions.get(rule, None), field, value)
   File "./cerberus/validator.py", line 1087, in _validate_allowed
     unallowed = set(value) - set(allowed_values)
 TypeError: unhashable type: 'dict'
```

expected behavior: a validation error, not a crash.

### PR DESCRIPTION
**Pull request by gilbsgilbs**
Fixes #524

### CODE REVIEW

**Comment by gilbsgilbs** [comment]
Failing test for py35 is unrelated to this changes FYI. I can reproduce a random fail on master with python 3.5.

File: cerberus/base.py [hunk 1]
```diff
@@ -1439,9 +1439,9 @@ def __validate_definitions(self, definitions, field):
     def _validate_allowed(self, allowed_values, field, value):
         """ {'type': 'container_but_not_string'} """
         if isinstance(value, Iterable) and not isinstance(value, str):
-            unallowed = set(value) - set(allowed_values)
+            unallowed = [v for v in value if v not in allowed_values]
```
Comment (funkyfuture) [inline cerberus/base.py]: i'd prefer:

- to store a tuple as error information with `tuple(<comprehension expression>)`
- use `x` inside the comprehension as `v` is commonly used when iterating over a mapping's values

**Comment by funkyfuture** [comment]
i'll take care of the doctest upon merging.

**Comment by funkyfuture** [comment]
amended, squashed and merged w/ bda3e7f11a1b1d052b8166637515a79740c83b13

**Pull request closed without merging**